akka-stream-tests/src/test/scala/akka/stream/scaladsl/QueueSourceSpec.scala

```diff
@@ -71,6 +71,14 @@ class QueueSourceSpec extends StreamSpec {
       f.futureValue should ===(QueueOfferResult.Enqueued)
     }
 
+    "reject elements when completed" in {
+      val (source, probe) = Source.queue[Int](42, OverflowStrategy.backpressure).delay(1000.seconds, DelayOverflowStrategy.backpressure).toMat(TestSink.probe)(Keep.both).run()
+      source.offer(42)
+      source.complete()
+      val f = source.offer(43)
```

Comment:
There is a race here, there is no guarantee that the stage did not already complete when we reach this line, and in that case the future is failed instead (because `invokeWithFeedback` fails the future if the stage is not running).

The only way I can think of to guarantee that this test always pass is to rather assert that `f` is either completed with `QueueOfferResult.Dropped` or a `StreamDetachedException`. 

Another idea would be to recover `StreamDetachedException` to a `QueueOfferResult.Dropped` but there may be user code relying on the future failing rather than completing for that case, so I'm not sure that is a good idea.

Reply:
I tried to avoid the race by introducing a delay into the stream. Don‘t you think this will work?

Reply:
Looking closer at the the stage, it won't complete until the buffer is empty, so if you had something backpressure that'd keep the `42` in the buffer. 

The delay operator won't backpressure until it's buffer is full though, so that won't do it. 

If you skip the delay and keep the source going directly into the sink, there is no demand signalled from the TestSink probe and the `42` will end up hanging around in the buffer so that when `Complete` comes in it won't complete the stage, only flag the stage as `terminating`, giving it time to complete the `43` offer with `Dropped`.

Maybe add a comment in the test explaining this, because it required a bit of hunting around in the code to figure this out.